src/rules/order.js

```diff
@@ -205,6 +269,7 @@ module.exports = {
       },
       'Program:exit': function reportAndReset() {
         makeOutOfOrderReport(context, imported)
+        makeOutOfAlphabeticalOrderReport(context, imported, sortOrder)
```

Comment:
Another separate thing:
I think it would be nice to to not re-report the modules that were reported in `makeOutOfOrderReport`.

Let's assume lowercase to be part of group 1, uppercase of group 2, and group 1 should appear before group 2.
```js
import 'B';
import 'A';
import 'b';
import 'a';
```
I haven't tried it yet, but I wouldn't be surprised if there were two error reports for moving B to a latter location or something. I would love some tests on that too, though the problem might not be shown with this example.

The reason I'm worried about this, is when you introduce the `order` rule to a new codebase, where imports are sorted in random ways, you'll get a lot of errors for the import section. I'd like for it to be the least confusing people and avoid having 20 errors when you required 10 modules. Note that the code might actually do this already, but without tests I have trouble being sure.

By not re-reporting modules, I think `findOutOfAlphabeticalOrder` could be simplified a bit too. I'm a bit confused about its inner workings at the moment.

Also, with only the `makeOutOfOrderReport` report, you have a consistent `should occur before`/`should occur after` message. I think it would be nice if that was true for all of the reports. Maybe compile the whole list with the group and the sort options one way, then do it the other way and take the shorter one.

Sorry to add a lot of work :sweat_smile: